apps/elasticsearch/templates/services/app/configs/readiness-probe.sh.j2

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+ES_HOST="127.0.0.1"
+ES_PORT="{{ elasticsearch_api_port }}"
+ES_STARTED="/tmp/es_started"
+
+
+testStatus() {
+    status_to_test="${1}"
+    expected_status="${2}"
+
+    declare -A statuses=( ["red"]=1 ["yellow"]=2 ["green"]=3)
+
+    if [[ -z "$status_to_test" || -z "${statuses[$status_to_test]}" ]] ; then
+        echo "Error : unknown status_to_test [$status_to_test]" 1>&2
+        exit 1
+    fi
+
+    if [[ -z "$expected_status" || -z "${statuses[$expected_status]}" ]]; then
+        echo "Error : unknown expected_status [$expected_status]" 1>&2
+        exit 1
+    fi
+
+    if [ ${statuses[$status_to_test]} -lt ${statuses[$expected_status]} ]; then
+        echo "Assertion failed : $status_to_test < $expected_status"
+        exit 1
+    fi
+}
+
+if [ -f "$ES_STARTED" ]; then
+    echo "Elasticsearch is already running. Checking if API is alive"
+    curl -fsSL "http://${ES_HOST}:${ES_PORT}/" > /dev/null
+    echo "OK"
+else
+    echo "Waiting for elasticsearch cluster to be running and in green state"
+    health_status=$(curl -fsSL "http://${ES_HOST}:${ES_PORT}/_cat/health?h=status" | sed -r 's/^[[:space:]]+|[[:space:]]+$//g')
+    testStatus "$health_status" "green"
+
+    echo "Waiting for initializing_shards to be 0"
+    init_shards=$(curl -fsSL "http://${ES_HOST}:${ES_PORT}/_cat/health?h=init" | sed -r 's/^[[:space:]]+|[[:space:]]+$//g')
+
+    if [ "${init_shards}" -gt 0 ]; then
+        echo "Synchronization not ready yet, initializing_shards = ${init_shards}"
+        exit 1
+    fi
+
+    touch "$ES_STARTED"
+
+    echo "Elasticsearch cluster is up (status = ${health_status})"
+fi
```

Comment:
I would suggest two things:

1. for variable references, use the `${myvar}` convention (for cross-project consistency, and I think it's more explicit)
2. use `shellcheck` (maybe it should be added to the CI) ; linting output on your script is:

```

In apps/elasticsearch/templates/services/app/configs/readiness-probe.sh.j2 line 26:
    if [ ${statuses[$status_to_test]} -lt ${statuses[$expected_status]} ]; then
         ^--------------------------^ SC2086: Double quote to prevent globbing and word splitting.
                                          ^---------------------------^ SC2086: Double quote to prevent globbing and word splitting.

For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```